modules/abstract-utxo/src/abstractUtxoCoin.ts

```diff
@@ -292,72 +292,21 @@ export abstract class AbstractUtxoCoin extends BaseCoin {
     return utxolib;
   }
 
-  /**
-   * Helper to get the version number for an address
-   */
-  protected getAddressVersion(address: string): number | undefined {
-    // try decoding as base58 first
-    try {
-      const { version } = utxolib.address.fromBase58Check(address, this.network);
-      return version;
-    } catch (e) {
-      // try next format
-    }
-
-    // if coin does not support script types with bech32 encoding, do not attempt to parse
-    if (!this.supportsAddressType('p2wsh') && !this.supportsAddressType('p2tr')) {
-      return;
-    }
-
-    // otherwise, try decoding as bech32
-    try {
-      const { version, prefix } = utxolib.address.fromBech32(address);
-      if (_.isString(this.network.bech32) && prefix === this.network.bech32) {
-        return version;
-      }
-    } catch (e) {
-      // ignore errors, just fall through and return undefined
-    }
-  }
-
-  /**
-   * Helper to get the bech32 prefix for an address
-   */
-  protected getAddressPrefix(address: string): string | undefined {
-    // otherwise, try decoding as bech32
-    try {
-      const { prefix } = utxolib.address.fromBech32(address);
-      return prefix;
-    } catch (e) {
-      // ignore errors, just fall through and return undefined
-    }
-  }
-
   /**
    * Check if an address is valid
    * @param address
    * @param forceAltScriptSupport
    */
   isValidAddress(address: string, forceAltScriptSupport = false): boolean {
```

Comment:
let's put support for non-defined formats behind a parameter

```suggestion
  isValidAddress(address: string, param?: { anyFormat: boolean } | /* legacy parameter */ boolean): boolean {
    if (typeof param === 'boolean' && param) {
      throw new Error('deprecated');
    }
    
    const formats = (param && param.anyFormat) ? ['default'] : undefined;
    
     try {
        utxolib.addressFormat.toOutputScriptTryFormats(address, this.network, formats);
        return true;
      } catch (e) {
        return false;
      }
  }  
```